src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,142 @@
+        testUserNamesake.setPassword("1234568");
+        try {
+            registrationService.register(testUserOk);
+            registrationService.register(testUserNamesake);
+        } catch (AuthenticationException e) {
+            return;
+        }
+        fail("AuthenticationException should be thrown because User has already existed");
+    }
+
+    @Test
+    void userAgeLessThanLimit_NotOk() {
+        User testUserJuvenile = new User(); //testUserJuvenile
+        testUserJuvenile.setId(12347L);
+        testUserJuvenile.setAge(16);
+        testUserJuvenile.setLogin("userJuvenile");
+        testUserJuvenile.setPassword("1234568");
+        try {
+            registrationService.register(testUserJuvenile);
+        } catch (AuthenticationException e) {
+            return;
+        }
+        fail("AuthenticationException should be thrown because User younger than 18");
+    }
+
+    @Test
+    void passwordLessThanLimit_NotOk() {
+        User testUserShortPassword = new User(); //testUserShortPassword
+        testUserShortPassword.setId(12348L);
+        testUserShortPassword.setAge(20);
+        testUserShortPassword.setLogin("userShortPassword");
+        testUserShortPassword.setPassword("12345");
+        try {
+            registrationService.register(testUserShortPassword);
+        } catch (AuthenticationException e) {
+            return;
+        }
+        fail("AuthenticationException should be thrown because Password shorter than 6 characters");
+    }
+
+    @Test
+    void registerNoId_NotOk() {
+        User testUserNoId = new User(); //testUserNoId
+        testUserNoId.setAge(20);
+        testUserNoId.setLogin("userNoId");
+        testUserNoId.setPassword("123456");
+        try {
+            registrationService.register(testUserNoId);
+        } catch (AuthenticationException e) {
+            return;
+        }
+        fail("AuthenticationException should be thrown because ID can't be null");
+    }
+
+    @Test
+    void registerNoAge_NotOk() {
+        User testUserNoAge = new User(); //testUserNoAge
+        testUserNoAge.setId(12348L);
+        testUserNoAge.setLogin("userNoAge");
+        testUserNoAge.setPassword("123456");
+        try {
+            registrationService.register(testUserNoAge);
+        } catch (AuthenticationException e) {
+            return;
+        }
+        fail("AuthenticationException should be thrown because Age can't be null");
+    }
+
+    @Test
+    void registerNoLogin_NotOk() {
+        User testUserNoLogin = new User(); //testUserNoLogin
+        testUserNoLogin.setId(12348L);
+        testUserNoLogin.setAge(20);
+        testUserNoLogin.setPassword("123456");
+        try {
+            registrationService.register(testUserNoLogin);
+        } catch (AuthenticationException e) {
+            return;
+        }
+        fail("AuthenticationException should be thrown because Login can't be null");
+    }
+
+    @Test
+    void registerNoPassword_NotOk() {
```

Comment:
<img width="606" alt="image" src="https://user-images.githubusercontent.com/81781479/228307872-3d827916-84d1-42d4-b0e6-dea9618bd036.png">

ofllow the pattern - you have not fixed it from the first review